extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/discovery/TradfriDiscoveryParticipant.java

```diff
@@ -51,8 +61,7 @@ public String getServiceType() {
     public ThingUID getThingUID(ServiceInfo service) {
         if (service != null) {
             String name = service.getName();
-            if ((service.getType() != null) && service.getType().equals(getServiceType())
-                    && (name.matches("gw:([a-f0-9]{2}[-]?){6}"))) {
+            if (Objects.equals(service.getType(), getServiceType()) && (name.matches(GATEWAY_NAME_REGEX_PATTERN))) {
```

Comment:
As mentioned previously, the check for the service type does not really belong here. I have created https://github.com/eclipse/smarthome/pull/4506 instead.

Reply:
As it is already available in the code - I don't want to introduce a regression - only wanted to reduce the length of the code - so that it fits in one line

Reply:
You're right, sorry. But as #4506 has been merged, I would suggest to remove this check here now in any case.